src/basicfuns.jl

```diff
@@ -246,6 +249,66 @@ function logsumexp(X::AbstractArray{T}; dims=:) where {T<:Real}
     end
 end
 
+"""
+    logsumexp_onepass(X)
+
+Compute `log(sum(exp, X))` in a numerically stable way that avoids intermediate under- and
+overflow.
+
+In contrast to [`logsumexp`](@ref) the result is computed using a single pass over the data.
+`X` should be an iterator of real numbers.
+
+# References
+
+[Sebastian Nowozin: Streaming Log-sum-exp Computation.](http://www.nowozin.net/sebastian/blog/streaming-log-sum-exp-computation.html)
+"""
+function logsumexp_onepass(X)
+    # fallback for empty collections
+    isempty(X) && return log(sum(X))
+
+    # perform single pass over the data
```

Comment:
I think as this point it's clear enough that this function does a single pass. :-)
```suggestion
```